### PR TITLE
 DCOS-12427: Display a helper message when service endpoints are not available

### PR DESCRIPTION
**Pull request by mlunoe**
Original implementation caused quite some confusion https://github.com/dcos/dcos-ui/pull/1603

The decision was made to display Service Endpoints Section disabled with a helper message.

Normal behavior:
screen shot 2017-01-04 at 17 56 32

User network:
screen shot 2017-01-04 at 17 56 41

Best,
@nLight

---

---
ℹ️ This PR introduced hiding the endpoints when runtime was `NONE` (aka. Mesos) and Network type was `USER` (aka. Virtual Network): https://github.com/dcos/dcos-ui/pull/1603

---

This PR disables the `USER` (aka, Virtual Network) selection in the dropdown when `NONE` (aka. Mesos) runtime is selected.

It also cleans up the two endpoint sections a bit, as they were very repetitive.
Here are the difference between the virtual network section and host:

Virtual network section and bridge:

Virtual network section and host for pods:

### CODE REVIEW

**Comment by orlandohohmeier** [comment]
@mlunoe why are we changing the current behavior? @nLight added the option to select a user network after quite some research and discussion with the core and networking team. Let's add a description text explaining why one can't select endpoints instead of removing the option to fix the UX issue documented in `DCOS-12427`.

**Comment by mlunoe** [comment]
@orlandohohmeier I don't understand why this was just left blank? Could you or @nLight please explain it in detail what is exptected and why? Clearly this is not what @leemonore expected, and neither did I

**Comment by orlandohohmeier** [comment]
@mlunoe I have to admit that leaving that section blank was a bad decision and we should change that. @nLight and I will work on a better solution next week.

**Comment by mlunoe** [comment]
What about the cleanup from this PR. I think that should still be a nice thing to have?

**Comment by orlandohohmeier** [comment]
@mlunoe we will work off your branch so that we don't lose the cleanup.

**Comment by nLight** [comment]
@mlunoe @leemunroe please have a look

File: plugins/services/src/js/components/forms/NetworkingFormSection.js [hunk 1]
```diff
@@ -471,17 +348,18 @@ class NetworkingFormSection extends mixin(StoreMixin) {
...
     const type = findNestedPropertyInObject(container, 'type');
-    let network = findNestedPropertyInObject(this.props.data, 'networkType');
+    const network = findNestedPropertyInObject(this.props.data, 'networkType');
     const disabledMap = {};
 
     // Runtime is Mesos
     if (!type || type === NONE) {
-      disabledMap[Networking.type.BRIDGE] = 'BRIDGE networking is not compatible with the Mesos runtime';
+      disabledMap[BRIDGE] = 'BRIDGE networking is not compatible with the Mesos runtime';
+      // disabledMap[USER] = 'Virtual networking is not compatible with the Mesos runtime';
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/NetworkingFormSection.js]: Let's remove this guy then

**Comment by leemunroe** [comment]
Message looks good.

To help improve immediate clarity I would like to see a more obvious disabled state and also the tooltip to be triggered when hovering over the whole container for that area i.e. not just the lock icon

More like this (opacity .5)
image

Tooltip trigger area
image

**Comment by nLight** [comment]
Review screen has to be fixed. Right now it displays: HOST for a USER network in Mesos Runtime.

**Comment by nLight** [comment]
Fixed the Review screen and improved 'hiding the networkName field' logic introduced in https://github.com/dcos/dcos-ui/pull/1701

**Comment by nLight** [comment]
ping @wavesoft

**Comment by nLight** [comment]
run unit tests

**Comment by leemunroe** [comment]
@nLight Better :)

Tracking the request for tooltips on blocks in DCOS-12589

**Comment by wavesoft** [comment]
LGTM! :shipit:

File: plugins/services/src/js/service-configuration/ServiceNetworkingConfigSection.js [hunk 2]
```diff
@@ -16,10 +16,25 @@ class ServiceNetworkingConfigSection extends ServiceConfigBaseSectionDisplay {
   /**
    * @override
    */
+  shouldExcludeItem(row) {
+    if (row.key !== 'ipAddress.networkName') {
+      return false;
+    }
+
+    const {appConfig} = this.props;
+    const networkName = findNestedPropertyInObject(appConfig,
```
Comment (mlunoe) [inline plugins/services/src/js/service-configuration/ServiceNetworkingConfigSection.js]: #nit This formatting is a little weird

Comment (mlunoe) [inline plugins/services/src/js/service-configuration/ServiceNetworkingConfigSection.js]: fixed

**Pull request merged**